### PR TITLE
Python

### LINKED ISSUE #2: Port to Python
This will allow us to use `usaddress` to parse single line inputs.

### LINKED ISSUE #3: Point at AGOL rather than the WebAPI for AddressPoints data

### PR DESCRIPTION
**Pull request by stdavis**
This is pretty much a direct port of the `proof-of-concept` branch from JS to Python.

This does point at AGOL for the address point data for the `/suggest` endpoint, but I think that performance is going to demand that we go a different route. AGOL feature service queries are too slow for a typeahead. I wondered if querying the Open SGID might be worth a try.

I've deployed this to my poc GCP project and configured [this web application](https://utah.maps.arcgis.com/apps/webappviewer/index.html?id=97a1529c31c84a93956968d48c6e08ad) to test it. Make sure to use the bottom, right search widget:
image

Closes #3 
Closes #2

### CODE REVIEW

**Comment by stdavis** [comment]
> Have we ever looked into the [premium data store option](https://doc.arcgis.com/en/arcgis-online/reference/faq.htm#DATA_STORE) from Esri?

Not that I know of. Although, I'm more interested in a non-Esri solution as a first option...

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from glob import glob
+from os.path import basename, splitext
+
+from setuptools import find_packages, setup
+
+setup(
+    name='masquerade',
+    version='1.0.0',
+    license='MIT',
+    description='Project description.',
```
Comment (steveoh) [inline setup.py:16]: **suggestion**: add a one liner

**Comment by stdavis** [comment]
> I bet if we [created an index](https://developers.arcgis.com/rest/services-reference/add-to-definition-feature-layer-.htm#example2) on `FullAdd` the queries would go much faster.

Oh sweet! I didn't know that you could do that. Any issues with me adding one?

File: setup.py (hunk 1 above)
Comment (stdavis) [reply setup.py:16]: Nice catch.

**Comment by steveoh** [comment]
No please do. We should consider doing it for more datasets.

**Comment by stdavis** [comment]
FYI - I was able to add an index on the `FullAdd` field for the address points hosted feature layer using [this code example](https://github.com/mpayson/esri-partner-tools/blob/master/feature_layers/manage_indexes.ipynb) and the performance is much better in the [test app](https://utah.maps.arcgis.com/apps/webappviewer/index.html?id=97a1529c31c84a93956968d48c6e08ad). It's still not lightning fast but much better.

**Pull request merged**